### PR TITLE
add recreate for myServices.

### PR DESCRIPTION
**Pull request by Alice1319**
fixes https://github.com/eclipse-xpanse/xpanse/issues/1988

recreate btn:
image

recreating:
Snipaste_2024-11-14_20-02-05-recreating

recreateSuccess:
Snipaste_2024-11-14_11-39-00-recreate-success

recreate destroy failed:
Snipaste_2024-11-14_19-46-49--recreate-destroy-failed

recreate deploy failed:
Snipaste_2024-11-14_20-06-09-recreate-failed-alert

### CODE REVIEW

**Comment by dengyali0125** [comment]
LGTM

File: src/components/content/common/deployedServiceDetailsByServiceIdQuery/useServiceDetailsByServiceIdQuery.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import {
+    getSelfHostedServiceDetailsById,
+    GetSelfHostedServiceDetailsByIdData,
+    getVendorHostedServiceDetailsById,
+    GetVendorHostedServiceDetailsByIdData,
+    serviceHostingType,
+    taskStatus,
+} from '../../../../xpanse-api/generated';
+
+export function useServiceDetailsByServiceIdQuery(
```
Comment (swaroopar) [inline src/components/content/common/deployedServiceDetailsByServiceIdQuery/useServiceDetailsByServiceIdQuery.ts:16]: the same query there here as well - https://github.com/eclipse-xpanse/xpanse-ui/blob/accd7dde31c42d75a837e278ae8366e99419b457/src/components/content/order/orderStatus/useServiceDetailsPollingQuery.ts#L51

please consolidate.

File: src/components/content/common/deployedServiceDetailsByServiceIdQuery/useServiceDetailsByServiceIdQuery.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/*
```
Comment (swaroopar) [inline src/components/content/common/deployedServiceDetailsByServiceIdQuery/useServiceDetailsByServiceIdQuery.ts:1]: the folder naming structure is bit odd. I did not realize it in the previous PRs.
Could you please change it like this?

common/queries/$queryName.ts -> basically one file for each query. no need of separate folder for each query.

File: src/components/content/common/latestServiceOrderStatusQuery/useLatestServiceOrderStatusQuery.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ import {
 import { deploymentStatusPollingInterval } from '../../../utils/constants.tsx';
 
 export function useLatestServiceOrderStatusQuery(
-    orderId: string | undefined,
+    orderId: string | undefined | null,
```
Comment (swaroopar) [inline src/components/content/common/latestServiceOrderStatusQuery/useLatestServiceOrderStatusQuery.ts:15]: do we really need this change?

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: src/components/content/common/deployedServiceDetailsByServiceIdQuery/useServiceDetailsByServiceIdQuery.ts (hunk 1 above)
Comment (Alice1319) [reply src/components/content/common/deployedServiceDetailsByServiceIdQuery/useServiceDetailsByServiceIdQuery.ts:16]: consolidated.

File: src/components/content/common/deployedServiceDetailsByServiceIdQuery/useServiceDetailsByServiceIdQuery.ts (hunk 2 above)
Comment (Alice1319) [reply src/components/content/common/deployedServiceDetailsByServiceIdQuery/useServiceDetailsByServiceIdQuery.ts:1]: good advice, refactored.

File: src/components/content/common/latestServiceOrderStatusQuery/useLatestServiceOrderStatusQuery.ts (hunk 3 above)
Comment (Alice1319) [reply src/components/content/common/latestServiceOrderStatusQuery/useLatestServiceOrderStatusQuery.ts:15]: we don't need this,  removed.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**